app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuth.scala

```diff
@@ -128,29 +130,35 @@ trait HelpToSaveAuth extends AuthorisedFunctions with AuthRedirects {
   private type ValidOrMissingUserInfo[A] = ValidatedNel[MissingUserInfo, A]
 
   private def getUserInfo(nino:        String,
-                          name:        Name,
+                          name:        Option[Name],
                           email:       Option[String],
                           dob:         Option[LocalDate],
-                          itmpName:    ItmpName,
+                          itmpName:    Option[ItmpName],
                           itmpDob:     Option[LocalDate],
-                          itmpAddress: ItmpAddress): Either[MissingUserInfos, UserInfo] = {
+                          itmpAddress: Option[ItmpAddress]): Either[MissingUserInfos, UserInfo] = {
 
     val givenNameValidation: ValidOrMissingUserInfo[String] =
-      itmpName.givenName.orElse(name.name).filter(_.nonEmpty)
+      itmpName.flatMap(_.givenName).orElse(name.flatMap(_.name)).filter(_.nonEmpty)
         .toValidNel(MissingUserInfo.GivenName)
 
     val surnameValidation: ValidOrMissingUserInfo[String] =
-      itmpName.familyName.orElse(name.lastName).filter(_.nonEmpty)
+      itmpName.flatMap(_.familyName).orElse(name.flatMap(_.lastName)).filter(_.nonEmpty)
         .toValidNel(MissingUserInfo.Surname)
 
     val dateOfBirthValidation: ValidOrMissingUserInfo[LocalDate] =
       itmpDob.orElse(dob)
         .toValidNel(MissingUserInfo.DateOfBirth)
 
-    val addressValidation: ValidOrMissingUserInfo[ItmpAddress] = {
-      val lineCount = List(itmpAddress.line1, itmpAddress.line2, itmpAddress.line3, itmpAddress.line4, itmpAddress.line5)
-        .map(_.map(_.trim).filter(_.nonEmpty)).collect { case Some(_) ⇒ () }.length
-      if (lineCount < 2 || !itmpAddress.postCode.exists(_.trim.nonEmpty)) {
+    val addressValidation: ValidOrMissingUserInfo[Option[ItmpAddress]] = {
+      val lineCount = List(
+        itmpAddress.flatMap(_.line1),
+        itmpAddress.flatMap(_.line2),
+        itmpAddress.flatMap(_.line3),
+        itmpAddress.flatMap(_.line4),
+        itmpAddress.flatMap(_.line5)
```

Comment:
so many flatMaps ^^^^ 
any better way of doing this ?

Reply:
``scala
val lineCount = itmpAddress.fold(0){a => List(a.line1, ..., a.line5).map(_.map(_.trim).filter(_.nonEmpty)).collect { case Some(_) ⇒ () }.).length
```
or 
```scala
itmpAddress.fold(Invalid(NonEmptyList.of(MissingUserInfo.Contact){ a =>
  /** what we had before *//
}
```?

Reply:
thanks, I always forget the `fold` operation :) 
looks like Jackie has already merged this PR, let me make the fix in a different branch and PR